kolibri/core/auth/api.py

```diff
@@ -430,11 +432,19 @@ def perform_update(self, serializer):
             update_session_auth_hash(self.request, instance)
 
 
+class SanitizeInputsSerializer(serializers.Serializer):
+    username = serializers.CharField()
+    facility = HexOnlyUUIDField()
+
+
 class UsernameAvailableView(views.APIView):
     def post(self, request):
-        username = request.data.get("username")
-        facility_id = request.data.get("facility")
-
+        serializer = SanitizeInputsSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        if not serializer.is_valid():
```

Comment:
This second validity check for the serializer can be removed. If the validation fails, it will automatically raise an exception due to `serializer.is_valid(raise_exception=True)`, so the code inside this if block would not be executed.

Reply:
Sure @LianaHarris360. I'll update it.